### ISSUE TITLE
Some examples won't run on m1 macs

### ISSUE THREAD

**Issue by vultix** [djeedai/bevy_hanabi#110]
When running some examples, like the `spawn` example, I'm getting the following error:

```
2022-12-25T15:02:28.965779Z ERROR wgpu::backend::direct: Handling wgpu errors as fatal by default    
thread 'Compute Task Pool (0)' panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_texture
      note: label = `view_depth_texture`
    Dimension X value 2560 exceeds the limit of 2048
```

Looks like the problem is in this code:
```rust
    let mut options = WgpuSettings::default();
    let limits = WgpuLimits::downlevel_defaults();
    options.constrained_limits = Some(limits);
```

Examples should ideally work out of the box for everyone, so we might want to leave this (already option) section commented out or somehow detect the appropriate limits

**Comment by djeedai** [comment]
The `view_depth_texture` is not created by Hanabi; I think it's the standard depth texture for the main camera of Bevy. Can you please try some Bevy examples too with these settings? It looks like this is either a Bevy or `wgpu` issue.

Also, that one example specifically shows how to run on older hardware with lower limits. Your M1 mac is at the opposite side of the performance spectrum. There will always be someone with some hardware for which this or that doesn't work, it's impossible to provide universal support. So I'd rather have at least one example targeted at lower spec hardware for coverage reasons, rather than always assume high-spec ones in all examples. All other examples use the default "Desktop" profile which should require much more powerful hardware to run, and so people with older hardware might argue the opposite, that this code using `downlevel_defaults()` should be in _all_ examples. That being said, if this is really an issue we can investigate either a better error message or renaming the example to `lowspec.rs` or something more explicit maybe?

**Comment by vultix** [comment]
Yeah, perhaps there should be an issue on bevy's side to raise a better error message here.  The problem is that the `downlevel_defaults()` sets a texture limit lower than the screen resolution.

**Comment by djeedai** [comment]
I ended up disabling that downlevel option in the `spawn` example, I think you convinced me @vultix that examples should run out of the box. Fixed in #135.
